src/main/python/car_trecweb.py

```diff
@@ -20,22 +25,48 @@ def writer(p, fp):
     content += (u'<DOCHDR>\n')
     content += (u'\n')
     content += (u'</DOCHDR>\n')
+    if meta_dict and p.para_id in meta_dict:
+        content += (u'<TITLE>\n')
+        content += (meta_dict[p.para_id]["title"])
+        content += (u'\n</TITLE>\n')
+        content += (u'<TAG>\n')
+        content += (meta_dict[p.para_id]["tag"])
+        content += (u'\n</TAG>\n')
     content += (u'<BODY>\n')
     content += (text)
     content += (u'\n</BODY>\n')
     content += (u'</DOC>\n')
     fp.write(content)
+    
+def sanitize_string(s):
+    s = parse.unquote(s)
+    s = re.sub(r'\W+', ' ', s)
+    return s
+    
+def create_metadata_dict(tsvfile):
+    if tsvfile == None:
+        return {}
+    with open(args.metadata_file, 'r') as tsvfile:
+        reader = csv.reader(tsvfile, delimiter='\t')
+        metadata_dict = {}
+        for row in tqdm.tqdm(reader, desc='loading metadata lines'):
+            metadata_dict[row[0]] = {"title":sanitize_string(row[1]), "tag":sanitize_string(row[2])}

```

Comment:
Add some validation code here to check that each row has the required number of fields and they are in a reasonable format.  Rename 'tag' to be more descriptive.

Reply:
will rename to headings